lib/railway_ipc/message_consumption.ex

```diff
@@ -19,12 +22,29 @@ defmodule RailwayIpc.MessageConsumption do
   ]
 
   def process(payload, handle_module, exchange, queue, message_module) do
-    new(payload, handle_module, exchange, queue)
-    |> decode_message(message_module)
-    |> persist_message()
-    |> handle_message()
+    try do
+      {:ok, result} =
+        @repo.transaction(fn ->
```

Comment:
There is perhaps some weirdness here in that a module outside of persistence is accessing repo. I could get around that maybe by giving persistence the handle_claim function or something? I think this is OK for now, but just want to call that out in case others have thoughts.

Reply:
Yea, if it's not too big of a lift, I'd put it in the persistence layer. That said, it's fine. We're probably going to refactor this stuff in the future. (Famous last words)

Reply:
If it's alright, I think I would opt to skip it for now. This update has already taken longer than I planned and moving that into persistence would wreak havoc on the test suite right now.
And yes, I think when we have some refactor time, this will be on the list.